internal/internal_worker_base.go

```diff
@@ -310,6 +310,34 @@ func (bw *baseWorker) runPoller() {
 	}
 }
 
+func (bw *baseWorker) tryReserveSlot() bool {
+	if !bw.isWorkerStarted || bw.isStop() {
+		return false
+	}
+	// Reserve a executor slot via a non-blocking attempt to take a poller
+	// request entry which essentially reserves a slot
+	select {
+	case <-bw.pollerRequestCh:
+	default:
+		return false
+	}
+	return true
```

Comment:
```suggestion
	select {
	case <-bw.pollerRequestCh:
		return true
	default:
		return false
	}
```

(just a nit if it reads clearer to you)